accounts/service.go

```diff
@@ -529,11 +561,32 @@ func (s *InterceptorService) TrackPayment(id AccountID, hash lntypes.Hash,
 		return nil
 	}
 
-	// Okay, we haven't tracked this payment before. So let's now associate
-	// the account with it.
-	account.Payments[hash] = &PaymentEntry{
-		Status:     lnrpc.Payment_UNKNOWN,
-		FullAmount: fullAmt,
+	if !ok {
+		// In the rare case that the payment isn't associated with an
+		// account yet, we do that first, to ensure that we will track
+		// the payment in case the service is restarted. This can happen
+		// if the payment is a KeySend or SendToRoute payment.
+		// This is attempted even if the service is disabled, as we'll
+		// then track the payment on next startup.
+		account.Payments[hash] = &PaymentEntry{
+			Status:     lnrpc.Payment_UNKNOWN,
+			FullAmount: fullAmt,
+		}
+
+		err := s.store.UpdateAccount(account)
```

Comment:
Given that we already do exactly this call below, don't we just want to move the sending of the error to the err chan to the `UpdateAccount` call below?

Reply:
Sorry, good catch :)